src/controllers/revocation.ts

```diff
@@ -846,14 +846,21 @@ export class RevocationController {
 				} as UpdateUnencryptedStatusListUnsuccessfulResponseBody);
 			}
 
+			// get status list
+			const unencrypted = await identityServiceStrategySetup.agent.searchStatusList2021(
+				did,
+				statusListName,
+				DefaultStatusActionPurposeMap[statusAction]
+			);
+
 			// construct formatted response
 			const formatted = {
 				updated: true,
 				revoked: (result as BulkRevocationResult)?.revoked || undefined,
 				suspended: (result as BulkSuspensionResult)?.suspended || undefined,
 				unsuspended: (result as BulkUnsuspensionResult)?.unsuspended || undefined,
-				resource: result.statusList,
-				resourceMetadata: result.resourceMetadata,
+				resource: unencrypted.resource,
+				resourceMetadata: unencrypted.resourceMetadata,
```

Comment:
This broadly looks fine but I don't understand what was wrong with the original one, and why this should be different for unencrypted vs encrypted (which it seems to be like, at the moment). Swagger seems fine. Can you weigh in here as well please @Eengineer1?

Reply:
This is repetitive / redundant logic.

It might not be obvious for Abdulla that the underlying library logic might return its corresponding format + formatting that response is service / consumer logic according to requirements.

i.e.

- `statusList` is cast to `resource`.
- `revoked`, `suspended`, `unsuspended` are optional, non-intersecting properties, as per top-level action.

Reply:
Done! ✅